### PR TITLE
Issue #652: releasenotes-builder: Release notes for GitHub shows escape at-symbol

### LINKED ISSUE #652: releasenotes-builder: Release notes for GitHub shows escape at-symbol
https://github.com/checkstyle/checkstyle/pull/12007#issuecomment-1221431907

### PR DESCRIPTION
**Pull request by Rahulkhinchi03**
Fixes #652

### CODE REVIEW

File: releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java [hunk 1]
```diff
@@ -313,6 +313,10 @@ private static ReleaseNotesMessage createReleaseNotesMessage(int number, String
         titleField.setAccessible(true);
         titleField.set(issue, title);
 
+        final Field escapedTitleField = issue.getClass().getDeclaredField(" \"`@`\"");
```
Comment (romani) [inline releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java:316]: you need escape function to make code like:
```
escape(issue.getClass().getDeclaredField("title"))

```

**Review by romani — CHANGES_REQUESTED** [review]
item:

File: releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java (hunk 1 above)
Comment (Rahulkhinchi03) [reply releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java]: Changes Done

Comment (romani) [reply releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java]: Share prove of testing.

File: releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java [hunk 2]
```diff
@@ -312,6 +312,7 @@ private static ReleaseNotesMessage createReleaseNotesMessage(int number, String
         final Field titleField = issue.getClass().getDeclaredField("title");
         titleField.setAccessible(true);
         titleField.set(issue, title);
+        escape(issue, titleField);
```
Comment (romani) [inline releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java:315]: you should not escape all other formats, it is problem only for Github release page.

**Review by romani — CHANGES_REQUESTED** [review]
item:

File: releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java (hunk 2 above)
Comment (Rahulkhinchi03) [reply releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java]: Changes Done

Comment (Rahulkhinchi03) [reply releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java]: @romani ping

File: releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java [hunk 3]
```diff
@@ -313,13 +313,26 @@ private static ReleaseNotesMessage createReleaseNotesMessage(int number, String
         titleField.setAccessible(true);
         titleField.set(issue, title);
 
+        final Field escapedField = issue.getClass().getDeclaredField("title");
+        escapedField.setAccessible(true);
+        escapedField.set(issue, title);
```
Comment (baratali) [inline releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java:318]: isn't it the same as `titleField` defined above?

Comment (Rahulkhinchi03) [reply releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java:318]: yes, the only difference is, that `titleField` is general and escapedField will be a field that escapes the `@` if found in the GitHub Page Notes.

File: releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java (hunk 1 above)
Comment (Rahulkhinchi03) [reply releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java]: I found something strange, the releasenotes-builder runs well with all other release numbers but when I run it for `10.1`,  it just runs and doesn't execute. I waited for a whole day for it to run but it just runs and shows no output ( for eg: `execution succeeded` doesn't come)

File: releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java (hunk 3 above)
Comment (baratali) [reply releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java:318]: I mean you update the same field twice with the same value:
```java
        final Field titleField = issue.getClass().getDeclaredField("title");
        titleField.setAccessible(true);
        titleField.set(issue, title);

        final Field escapedField = issue.getClass().getDeclaredField("title");
        escapedField.setAccessible(true);
        escapedField.set(issue, title);
```

**Review by romani — CHANGES_REQUESTED** [review]
please add unit test that have in data `@` symbol and validate that resulted generated text have this symbols escaped in Github release notes and NOT escaped in XML(Xdoc) release notes.

File: releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java (hunk 1 above)
Comment (romani) [reply releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java]: please focus on writing junit test to test your code, you can prove all execution correctness by junit only.
and later to execution on real release notes.

File: releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java (hunk 3 above)
Comment (Rahulkhinchi03) [reply releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java:318]: I am not sure if this is what @romani wants me to do. Please confirm that do I need to create an another value or title is good.

Comment (romani) [reply releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java:318]: Yes, you need to create new template field, that is used only in GitHub template.

**Comment by rnveach** [comment]
Release notes builder was updated (fixes and Java 11) and TC is back to requiring no errors.
This PR should be rebased to ensure there are no issues.

**Pull request closed without merging**